pytential/symbolic/mappers.py

```diff
@@ -280,7 +280,12 @@ def map_common_subexpression(self, expr):
 # {{{ FlattenMapper
 
 class FlattenMapper(FlattenMapperBase, IdentityMapper):
-    pass
+    def map_int_g(self, expr):
+        densities, kernel_arguments, changed = rec_int_g_arguments(self, expr)
+        if not changed:
+            return expr
+
+        return replace(expr, densities=densities, kernel_arguments=kernel_arguments)
```

Comment:
Also from #162, it would be nice to teach `ExpressionKernel` in sumpy to flatten a bit more. Not sure what the best place for that is..